fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRunManager.java

```diff
@@ -57,16 +62,37 @@ public SubmitResult submitTaskRun(TaskRun taskRun, int priority) {
 
         String queryId = UUIDUtil.genUUID().toString();
         TaskRunStatus status = taskRun.initStatus(queryId, System.currentTimeMillis());
-        status.setPriority(priority);
+        status.setPriority(option.getPriority());
+        status.setMergeRedundant(option.isMergeRedundant());
         GlobalStateMgr.getCurrentState().getEditLog().logTaskRunCreateStatus(status);
-        long taskId = taskRun.getTaskId();
-
-        PriorityBlockingQueue<TaskRun> taskRuns = pendingTaskRunMap.computeIfAbsent(taskId,
-                u -> Queues.newPriorityBlockingQueue());
-        taskRuns.offer(taskRun);
+        arrangeTaskRun(taskRun, option.isMergeRedundant());
         return new SubmitResult(queryId, SubmitResult.SubmitStatus.SUBMITTED);
     }
 
+    // At present, only the manual and automatic tasks of the materialized view have different priorities.
+    // The manual priority is higher. For manual tasks, we do not merge operations.
+    // For automatic tasks, we will compare the definition, and if they are the same,
+    // we will perform the merge operation.
+    public void arrangeTaskRun(TaskRun taskRun, boolean mergeRedundant) {
+        if (!tryTaskRunLock()) {
+            return;
+        }
+        try {
+            long taskId = taskRun.getTaskId();
+            PriorityBlockingQueue<TaskRun> taskRuns = pendingTaskRunMap.computeIfAbsent(taskId,
+                    u -> Queues.newPriorityBlockingQueue());
+            if (mergeRedundant) {
+                // The remove here is actually remove the old TaskRun.
+                // Note that the old TaskRun and new TaskRun may have the same priority and definition,
+                // but other attributes may be different, such as creation time.
```

Comment:
Why don't keep only one task run with the highest priority if task runs have the same definition?